src/EditorFeatures/CSharpTest/AddDebuggerDisplay/AddDebuggerDisplayTests.cs

```diff
@@ -0,0 +1,367 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.AddDebuggerDisplay;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.AddDebuggerDisplay
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsAddDebuggerDisplay)]
+    public sealed class AddDebuggerDisplayTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+        {
+            return new CSharpAddDebuggerDisplayCodeRefactoringProvider();
+        }
+
+        [Fact]
+        public async Task OfferedOnEmptyClass()
+        {
+            await TestInRegularAndScriptAsync(@"
+[||]class C
+{
+}", @"
+using System.Diagnostics;
+
+[DebuggerDisplay(""{GetDebuggerDisplay(),nq}"")]
+class C
+{
+    private string GetDebuggerDisplay()
+    {
+        return ToString();
+    }
+}");
+        }
+
+        [Fact]
+        public async Task OfferedOnEmptyStruct()
+        {
+            await TestInRegularAndScriptAsync(@"
+[||]struct Foo
+{
+}", @"
+using System.Diagnostics;
+
+[DebuggerDisplay(""{GetDebuggerDisplay(),nq}"")]
```

Comment:
💭 I believe this form will result in a diagnostic saying `GetDebuggerDisplay` is unused. We could resolve that by using the following form instead:

```suggestion
[DebuggerDisplay(""{"" + nameof(GetDebuggerDisplay) + ""(),nq}"")]
```

Reply:
@sharwell Are you talking about the IDE diagnostic for unused members? If so, I think we do handle the original case: https://github.com/dotnet/roslyn/blob/7bc44488c661fd6bbb6c53f39512a6fe0cc5ef84/src/EditorFeatures/CSharpTest/RemoveUnusedMembers/RemoveUnusedMembersTests.cs#L1943-L1952

Reply:
Happy to make this change. I almost did this but wasn't sure because of how odd it looks. ReSharper also suggests that `nameof` should be used IIRC. You have a great point about it being easy to apply.

Reply:
Looks like I'll be doing this for C# only, since VB does not allow NameOf on private members from attribute arguments:

```vb
' BC30390 'Foo.Private Function GetDebuggerDisplay() As String' is not accessible in this
' context because it is 'Private'.
<DebuggerDisplay("{" + NameOf(Foo.GetDebuggerDisplay) + "(),nq}")>
Class Foo
    Private Function GetDebuggerDisplay() As String
        Return ToString()
    End Function
End Class
```